### PR TITLE
feat: add mutation testing with Gremlins

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

- Add Gremlins mutation testing to measure test effectiveness
- Configure weekly scheduled runs with PR diff-based testing option
- Set 60% mutation score threshold as quality gate
- Generate HTML/JSON reports as artifacts

## What is Mutation Testing?

Mutation testing measures test quality by introducing small changes (mutations) to the code and checking if tests detect them:

- **Killed mutants**: Tests caught the mutation (good!)
- **Survived mutants**: Tests missed the mutation (needs improvement)

A high mutation score indicates your tests are effective at catching bugs.

### Gremlins Configuration (`.gremlins.yaml`)
- **Mutators**: Boundary conditions, negations, increments, logical inversions
- **Coverage-aware**: Only mutates code covered by tests
- **Exclusions**: Test files, mocks, static assets
- **Threshold**: 60% mutation score (can be adjusted)

### GitHub Workflow (`.github/workflows/mutation.yml`)
- **Weekly schedule**: Runs Sunday 2 AM UTC
- **PR mode**: Diff-based testing on changed files only
- **Manual trigger**: Available with full/diff mode option
- **Artifacts**: JSON and HTML reports retained 30 days
- **PR comments**: Automatic results summary

### Why Gremlins?

| Feature | Gremlins | Alternatives |
|---------|----------|--------------|
| Last Update | Dec 2025 | 6+ months old |
| Coverage-Aware | Yes | No |
| CI Gates | Built-in | Manual |
| GitHub Action | Official | None |

## Test plan

- [ ] Verify workflow syntax with actionlint
- [ ] Run mutation tests locally with `gremlins unleash`
- [ ] Confirm scheduled runs trigger correctly
- [ ] Verify PR comments appear on mutation runs

### CODE REVIEW

File: .github/workflows/mutation.yml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  # Run on PRs for diff-based mutation testing (optional gate)
+  pull_request:
+    branches: [main]
+    paths:
+      - '**.go'
+      - '.gremlins.yaml'
+      - '.github/workflows/mutation.yml'
+
+permissions:
+  contents: read
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply .github/workflows/mutation.yml]: Fixed! Added `pull-requests: write` permission in commit 3224324.

File: .github/workflows/mutation.yml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+          go-version-file: go.mod
+
+      - name: Download dependencies
+        run: go mod download
+
+      - name: Install Gremlins
+        run: go install github.com/go-gremlins/gremlins/cmd/gremlins@latest
+
+      - name: Run mutation tests (full)
+        if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' && github.event.inputs.diff_only != 'true'
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply .github/workflows/mutation.yml]: Fixed! Added explicit parentheses for clarity in commit 3224324.

File: .github/workflows/mutation.yml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+          # Get changed Go files
+          CHANGED_FILES=$(git diff --name-only origin/main...HEAD -- '*.go' | grep -v '_test.go' || true)
+          if [ -z "$CHANGED_FILES" ]; then
+            echo "No Go source files changed, skipping mutation testing."
+            echo "## Mutation Testing Results" >> $GITHUB_STEP_SUMMARY
+            echo "No Go source files changed - skipped." >> $GITHUB_STEP_SUMMARY
+            exit 0
+          fi
+          echo "Testing mutations in: $CHANGED_FILES"
+          gremlins unleash --config=.gremlins.yaml --diff 2>&1 | tee mutation-output.txt
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply .github/workflows/mutation.yml]: Fixed! Removed the unused `CHANGED_FILES` variable and simplified to use inline `git diff` check directly. The gremlins `--diff` flag handles the actual diff detection internally.

File: .gremlins.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+# Diff mode: only test mutations in changed files (for CI efficiency)
+# Set via CLI flag --diff for PR workflows
+diff: false
+
+# Tags for conditional compilation
+tags: ""
+
+# Additional test flags
+test-cpu: 0
+test-timeout: 60s
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply .gremlins.yaml]: Good point\! I'll create a follow-up issue to add mutation testing documentation to `docs/TESTING.md` covering local setup, running tests, and interpreting results. This PR focuses on the infrastructure - documentation can be added separately.

[the remark this answers is not in the document]
Comment (CybotTM) [reply .gremlins.yaml]: Fixed in commit c3b050a - added comprehensive mutation testing documentation to docs/tests.md covering:
- What mutation testing is and its benefits
- How to run Gremlins locally
- Configuration options
- CI integration details
- Tips for improving mutation score
